Add a search engine for core.ac.uk

return42 opened the pull request:
## What does this PR do?

Add a search engine for core.ac.uk : https://core.ac.uk/

Initial commit b416231 was fetched-from: https://github.com/searx/searx/pull/2688 / for  the rest see commit messages.

marc-cenon committed:
add core.ac.uk engine

Commit:
[mod] core.ac.uk engine

- add to list of pylint scripts
- add debug log messages
- move API key int `settings.yml`
- improved readability
- add some metadata to results

Signed-off-by: Markus Heiser <markus@darmarit.de>

dalf commented on searx/engines/core.py:
```diff
@@ -0,0 +1,82 @@
+
+about = {
+    "website": 'https://core.ac.uk',
+    "wikidata_id": None,
```
The wikidata ID is Q22661180

return42 replied:
Thanks!  .. done.

dalf commented on searx/engines/core.py:
```diff
@@ -0,0 +1,82 @@
+paging = True
+nb_per_page = 10
+
+api_key = 'unset'
```
If a key is required, the engine definition in settings.yml should be in comments

return42 replied:
Ooops .. I forgot .. done!

dalf commented on searx/settings.yml:
```diff
@@ -273,6 +273,15 @@ engines:
+    # api_key : "xxxxxxxx"
+    # set api_key and comment out disabled ..
+    disabled: True
+
```
It is the whole engine definition that should be comment out : without the api_key nothing works AFIK.

Commit:
[fix] Springer-Nature & core.ac need API key to work

It is the whole engine definition that should be comment out : without the
api_key nothing works.

Signed-off-by: Markus Heiser <markus@darmarit.de>
